### DISCUSSION TITLE
Force Alignment in TTS

### DISCUSSION THREAD

**Discussion by nellorebhanuteja** [espnet/espnet#4383] [Q&A] [answered]
FastSpeech2 uses Montreal Force Aligner (MFA) for learning duration model.
In ESPnet implementation, the time stamps are obtained using Tacotron2.
Is there a reason for this?

Does ESPnet support training using force alignment tools such as MFA/Pytorch Wav2Vec based Force Aligner ?

@kan-bayashi

**Comment by kan-bayashi** [comment, marked as answer]
> In ESPnet implementation, the time stamps are obtained using Tacotron2.

Tacotron 2 or Transformer-TTS

> Is there a reason for this?

This is because it is difficult to support various languages when using MFA.
And we need to match the phoneme representation with ASR models, which is painful.
Furthermore, we cannot use extra symbols which do not appear in ASR.
In data-driven manner, we do not need to consider the above points.

> Does ESPnet support training using force alignment tools such as MFA/Pytorch Wav2Vec based Force Aligner ?

If you create `durations` file, you can do it.
Recently we support JETS, which is joint training of FS2 and HiFiGAN without GT durations.
You can try it (See https://github.com/espnet/espnet/pull/4364).

**Reply by nellorebhanuteja** [reply]
Got it.
Thank you!
